app/javascript/components/displays.jsx

```diff
@@ -0,0 +1,214 @@
+  const { enabled: enabledWidgets, byIdOrFirst: current } = widgets;
+
+  const WidgetElement = widgetElements[current.name].Panel;
+
+  const index = enabledWidgets.findIndex(w => w.id === current.id);
+
+  const prevWidgetId =
+    enabledWidgets[mathMod(index - 1, enabledWidgets.length)].id;
+
+  const nextWidgetId =
+    enabledWidgets[mathMod(index + 1, enabledWidgets.length)].id;
+
+  const setNextWidgetId = () => {
+    stateCallback(({ transitionTime, pausedTime }) => ({
+      currentWidgetId: nextWidgetId,
+      transitionTime,
+      pausedTime,
+    }));
+  };
+
+  const setPrevWidgetId = () => {
+    stateCallback(({ transitionTime, pausedTime }) => ({
+      currentWidgetId: prevWidgetId,
+      transitionTime,
+      pausedTime,
+    }));
+  };
+
+  const switchPages = ({ keyCode }) => {
+    const upKey = 38;
+    const downKey = 40;
+    const SPACEBAR = 32;
+    const { pausedTime } = superState;
+    if (keyCode === SPACEBAR) {
+      if (pausedTime) {
+        stateCallback(({ currentWidgetId, transitionTime }) => ({
+          currentWidgetId,
+          pausedTime: null,
+          transitionTime: transitionTime - pausedTime + Date.now(),
+        }));
+      } else {
+        stateCallback(({ currentWidgetId, transitionTime }) => ({
+          currentWidgetId,
+          transitionTime,
+          pausedTime: Date.now(),
+        }));
+      }
+    }
+    if (keyCode === downKey) {
+      setNextWidgetId();
+    }
+    if (keyCode === upKey) {
+      setPrevWidgetId();
+    }
+  };
+
+  const handleClicks = ({ target }) => {
+    const targetId = target.id;
+    if (targetId.startsWith('widget_')) {
+      const widgetId = parseInt(targetId.replace('widget_', ''), 10);
+      stateCallback({
+        currentWidgetId: widgetId,
+      });
+    }
+  };
+
+  useEffect(() => {
+    timerId.current = setInterval(() => {
+      const { transitionTime, pausedTime } = superState;
+      if (!transitionTime || pausedTime) {
+        return;
+      }
+      const prefetchTime = transitionTime - prefetchDelay;
+      const now = Date.now();
+
+      if (transitionTime < now) {
+        setNextWidgetId();
+      } else if (
+        prefetchTime < now &&
+        lastPrefetchTime.current < now - prefetchDelay
+      ) {
+        lastPrefetchTime.current = now;
+        setNextWidgetId();
+      }
+    }, 500);
+
+    return () => {
+      if (timerId.current) {
+        clearInterval(timerId.current);
+      }
+    };
+  }, [superState]);
+
+  return (
+    <Wrapper
+      onKeyDown={switchPages}
+      onClick={handleClicks}
+      // eslint-disable-next-line
+      tabIndex="0"
+    >
+      <FullPanel>
+        <WidgetElement cityName={cityName} />
+      </FullPanel>
+      <SidePanel
+        widgets={enabledWidgets}
+        selectedWidgetId={superState.currentWidgetId}
+        showWeather={current.showWeather}
+        totalTime={current.durationSeconds * 1000}
+        isPaused={!!superState.pausedTime}
```

Comment:
Why do you have !!superState?

Reply:
The field isPaused checks whether the Widget timer is currently paused for rendering purposes. !!superState.pausedTime casts superState.pausedTime to a boolean value (from either null, <a_number> to false, true respectively)